File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = [0] * len(prices)
    n = len(prices)
    stack = []
    
    for i in range(n):
        while stack != [] and stack[-1][1] > prices[i]:
            pre, _ = stack.pop()
            answer[pre] = i - pre
        stack.append((i, prices[i]))
    
    for i, s in stack:
        answer[i] = n - 1 - i  
    
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    var answer = [];
    for(let i=0; i<prices.length-1; i++){
        let time = 0;
        let a = prices[i];
        for (let j=i+1; j<prices.length; j++){
            time++
            if (a <= prices[j]) {
                if (j===prices.length-1) {
                    answer.push(time)
                    break
                }
            }else{
                answer.push(time)
                break
            }
        }
    }
    answer.push(0)
    return answer;
}